application/src/main/java/io/redlink/smarti/webservice/RocketChatEndpoint.java

```diff
@@ -213,6 +210,35 @@ private String toJsonString(SmartiUpdatePing smartiUpdatePing) throws JsonProces
         }
     }
 
+    /**
+     * Allow conversation-independent search.
+     * @param clientName the client id
+     * @param queryParams the actual query-params
+     */
+    @ApiOperation(value = "search for a conversation", response = SearchResult.class,
+            notes = "besides simple text-queries, you can pass in arbitrary solr query parameter.")
+    @RequestMapping(value = "{clientId}/search", method = RequestMethod.GET)
+    public ResponseEntity<?> search(
+            @PathVariable(value = "clientId") String clientName,
+            @ApiParam("fulltext search") @RequestParam(value = "text", required = false) String text,
+            @ApiParam(hidden = true) @RequestParam MultiValueMap<String, String> queryParams) {
```

Comment:
so I can use any of the supported search query parameters of Solr by simply appending these to the HTTP query  string?

Reply:
Yes, but:

* the required filters for access-controll (`fq=owner:<clientId>`) and type (`fq=type:conversation`) are always appended.
* if a `text`-parameter is provided, any `q`-parameter passed in is **replaced** by `q={!parent which="type:conversation"}text:<text>`